### PR TITLE
Support array configuration values [14957]

### PR DESCRIPTION
**Pull request by jparisu**

### CODE REVIEW

File: plugins/datastreamer_plugin/utils/dynamic_types_utils.cpp [hunk 1]
```diff
@@ -95,6 +96,27 @@ void get_introspection_type_names(
...
+                    DEBUG(
+                        "Truncating array " << base_type_name <<
+                        " of size " << this_array_size <<
+                        " to size " << data_type_configuration.max_array_size);
+                    this_array_size = data_type_configuration.max_array_size;
+                }
+                // Could not be neither of them, it would be an inconsistency
+            }
+
             for (unsigned int i = 0; i < this_array_size; i++)
```
Comment (antonrv) [inline plugins/datastreamer_plugin/utils/dynamic_types_utils.cpp]: NIT: I know this does not belong to this commit (missed in the corresponding review), but I'd improve readability here, renaming `unsigned int i` to `MemberId member_id`, to express that the loop variable is actually a MemberId and not just a typical indexing variable

**Review by antonrv — DISMISSED** [review]
Good to me. Just a NIT comment (about code not belonging to PR), so approved anyway

**Comment by jparisu** [comment]
I have updated the way arrays are divided in time series using the std format of PlotJuggler:
Instead of:
```
datatype/points/0
datatype/points/1
```

use


```
datatype/points[]/points[0]
datatype/points[]/points[1]
```

**Review by antonrv — APPROVED** [review]
good to me
